tools/Raven.Debug/LogTrafficWatch/TrafficWatchReplay.cs

```diff
@@ -0,0 +1,313 @@
+                                using var response = await HttpClient.SendAsync(postRequest);
+                                await ConsumeResponse(response, responseBuffer);
+                                break;
+                            }
+                        }
+
+
+                    }
+                    catch (Exception ex)
+                    {
+                        Console.WriteLine(ex);
+                        // ignore
+                    }
+                }
+            }
+
+            private static async Task ConsumeResponse(HttpResponseMessage response, byte[] responseBuffer)
+            {
+                await using var stream = await response.Content.ReadAsStreamAsync();
+                while (await stream.ReadAsync(responseBuffer) > 0)
+                {
+                }
+            }
+
+            private Uri UriReplace(string original)
+            {
+                try
+                {
+                    return new UriBuilder(original) {
+                        Host = _host,
+                        Port = _port,
+                        Scheme = _schema
+                    }.Uri;
+
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e);
+                    return null;
+                }
+            }
+
+            private static string ConvertJsonElement(JsonElement e)
+            {
+                return e.ValueKind switch
+                {
+                    JsonValueKind.Number => e.ToString(),
+                    JsonValueKind.Array => e.ToString(),
+                    JsonValueKind.Object => e.ToString(),
+                    _ => $"\"{e}\""
+                };
+            }
+        }
+
+        public static int RegexIndexOf(string str, string pattern)
+        {
+            var m = Regex.Match(str, pattern);
+            return m.Success ? m.Index : -1;
+        }
+
+
+        private static void InitializeHttpClient(string certPath, string certPass)
+        {
+            if (string.IsNullOrEmpty(certPath) == false)
+            {
+                HttpClient = new HttpClient(new HttpClientHandler
+                {
+                    ClientCertificates =
+                    {
+                        new X509Certificate2(certPath, certPass, X509KeyStorageFlags.MachineKeySet)
+                    }
+                });
+                return;
+            }
+
+            HttpClient = new HttpClient();
+        }
+
+        private static void ExitAndPrintError(Exception ex, string errorText)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine(errorText);
+            Console.ResetColor();
+            if (ex != null)
+            {
+                Console.WriteLine();
+                Console.WriteLine(ex);
+            }
+
+            Environment.Exit(2);
+        }
+
+        public class TrafficWatchChange
```

Comment:
Is that possible to take that directly from raven?

Reply:
Done by adding [assembly: InternalsVisibleTo("Raven.Debug, PublicKey=xxx")] to ..src\Raven.Client\Properties\AssemblyInfo.Windows.cs and ..src\Raven.Client\Properties\AssemblyInfo.Linux.cs